src/goto-instrument/loop_utils.cpp

```diff
@@ -11,12 +11,35 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "loop_utils.h"
 
+#include <util/expr_util.h>
 #include <util/pointer_expr.h>
 #include <util/std_expr.h>
 
+#include <goto-programs/pointer_arithmetic.h>
+
 #include <analyses/natural_loops.h>
 #include <analyses/local_may_alias.h>
 
+class loop_utils_is_constantt : public is_constantt
+{
+public:
+  explicit loop_utils_is_constantt(const modifiest &mod) : modifies(mod)
+  {
+  }
+  bool is_constant(const exprt &expr) const override
```

Comment:
Minor query - I wonder if this would be better with a more explicit name, such as `is_constant_or_unmodified` ?

Reply:
I agree, that would be a better name. But the `operator()` defined on the base class (`is_constantt`) calls `is_constant` and so I chose to override the `is_constant` function.

Reply:
Yeah, I assumed it was a "more hassle than its worth" choice :-)